### PR TITLE
CORE-7032 Use batch send and receive in receive and finality flow

### PR DESCRIPTION
**Pull request by nkovacsx**
### Overview

Use batch sending and receiving in finality and receive finality flows. We used to send data one-by-one to counterparties but we can replace these with batch sending.

Performance testing: https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4504518727/Test+6+Use+batch+send+receive

### CODE REVIEW

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt [hunk 1]
```diff
@@ -77,8 +77,8 @@ class UtxoFinalityFlowV1(
 
     @Suspendable
     private fun sendTransactionAndBackchainToCounterparties() {
+        flowMessaging.sendAll(initialTransaction, sessions.toSet())
```
Comment (mnesbit) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:80]: Do we correctly handle empty session set? Perhaps just add an isEmpty check around this

**Review by rick-r3 — APPROVED** [review]

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt [hunk 2]
```diff
@@ -92,16 +92,17 @@ class UtxoFinalityFlowV1(
...
+            ).mapValues {
                 @Suppress("unchecked_cast")
-                session.receive(Payload::class.java) as Payload<List<DigitalSignatureAndMetadata>>
-            } catch (e: CordaRuntimeException) {
-                log.warn("Failed to receive signatures from ${session.counterparty} for transaction $transactionId")
-                persistInvalidTransaction(initialTransaction)
-                throw e
+                it.value as Payload<List<DigitalSignatureAndMetadata>>
             }
+        } catch (e: CordaRuntimeException) {
```
Comment (mnesbit) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:102]: There are possibly some non-CordaRuntimeExceptions possible here e.g. the cast could fail. So not sure if catch is too narrow, though the re-throw really needs to be a CordaRuntimeException. Probably not to be for tonight, but needs a JIRA to revisit

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt (hunk 1 above)
Comment (nkovacsx) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:80]: `sendAll` will not do anything if the sessions set is empty:
```
if (sessions.isEmpty()) {
            return
        }
```
Same with `toSet`, it's not sensitive to empty collections.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt (hunk 2 above)
Comment (nkovacsx) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:102]: The logic hasn't changed from the previous impl, this is how it was before. Probably need to reconsider.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt (hunk 1 above)
Comment (mnesbit) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:80]: OK

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt (hunk 2 above)
Comment (mnesbit) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:102]: revisit later

**Review by mnesbit — APPROVED** [review]
OK

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt (hunk 2 above)
Comment (nkovacsx) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt:102]: Raised https://r3-cev.atlassian.net/browse/CORE-14402

**Pull request merged**